pkg/client/restclient/config.go

```diff
@@ -62,6 +62,9 @@ type Config struct {
 	// TODO: demonstrate an OAuth2 compatible client.
 	BearerToken string
 
+	// Server requires plugin-specified authentication.
+	AuthPlugin string
```

Comment:
I'm not sure this ends up being enough structure for what we're likely to need.  It seems like the next likely step is needing to have a piece of information I need to pass through to my plugin.  Internally we wouldn't make it ahead of time, but you're plumbing this through a serialized file so I'm more concerned.